corehq/motech/repeaters/models.py

```diff
@@ -1574,6 +1574,36 @@ def try_now(self):
         # TODO rename to should_try_now
         return self.state != State.Success
 
+    def attempt_forward_now(self, *, is_retry=False, fire_synchronously=False):
+        from corehq.motech.repeaters.tasks import (
+            process_repeat_record,
+            retry_process_repeat_record,
+        )
+
+        if self.next_check is None or self.next_check > datetime.utcnow():
+            return
+
+        # Set the next check to happen an arbitrarily long time from now.
+        # This way if there's a delay in calling `process_repeat_record` (which
+        # also sets or clears next_check) we won't queue this up in duplicate.
+        # If `process_repeat_record` is totally borked, this future date is a
+        # fallback.
+        updated = type(self).objects.filter(
```

Comment:
> if we ever change `SQLRepeatRecord` to `RepeatRecord`

This is expected to happen after the Couch model is removed.

Otherwise, no significant advantage. Personally I don't like referencing model names explicitly in methods.